source/docs/casper/resources/tutorials/advanced/cross-contract.md

```diff
@@ -0,0 +1,538 @@
+extern crate alloc;
+
+// the elementary types 
+use alloc::string::String;
+use alloc::vec::Vec;
+use alloc::collections::BTreeMap;
+use crate::alloc::string::ToString;
+
+
+// casper crates
+use casper_types::{Key, CLType, Parameter, EntryPoint, EntryPoints, EntryPointType, EntryPointAccess};
+
+use casper_contract::{
+    contract_api::{runtime, storage},
+    unwrap_or_revert::UnwrapOrRevert,
+};
+
+
+#[no_mangle]
+pub extern "C" fn call() {
+
+    // get the value of a passed parameter with the key "message"
+    let value: String = runtime::get_named_arg("message");
+
+    // the value will be wraped in a URef
+    let value_ref = storage::new_uref(value);
+
+    // creating the new set of named keys
+    // the keys are a Map of Key/Value 
+    let mut named_keys: BTreeMap<String, Key> = BTreeMap::new();
+
+    // insert the new value into the named keys
+    named_keys.insert(String::from("message"),value_ref.into()); // use into to wrap the value to the key
+    // create a new vector 
+    let mut vec = Vec::new();
+    vec.push(Parameter::new("message", CLType::String));
+
+    // create an Entry Point Object
+    let mut entry_points = EntryPoints::new();
+
+    // add the entry point to the entry points object
+    entry_points.add_entry_point(EntryPoint::new(
+        "update_msg",                   // the name of the entry point
+        vec,                            // the arguments which can be passed into the entry point
+        CLType::Unit,                   // return type of the entry point
+        EntryPointAccess::Public,       // acces permissions - public can be accessed always
+        EntryPointType::Contract        // in most cases it will be contract
+    ));
+
+    // the contract is stored in the global state
+    let (stored_contract_hash, _contract_version) = storage::new_contract(
+        entry_points,                                       // entry points
+        Some(named_keys),                                   // named keys 
+        Some("Hello_world_package_name".to_string()),       // package name
+        Some("Hello_world_access_uref".to_string())         // access uref
+    );
+
+    // to access from the account - named keys of the account
+    runtime::put_key("hello_world_contract", stored_contract_hash.into());
+
+}
+```
+
+:::tip
+
+We should notice that that runtime and storage appear quite frequently in our code. If you didn't have a chance to check out those modules yet it would be a perfect time to do so: [contract API](https://docs.rs/casper-contract/latest/casper_contract/contract_api/index.html).
+The amount of native functions isn't that large so it might be advisable to think about which one of those could be used with your scenario.
+
+:::
+
+You should always keep the entry points and the functions of the contract in sync. If there is an entry point with name `update_msg` there should be also a function with the same name in the contract.
+
+This is presented below:
```

Comment:
This is another vague sentence that could technically refer to a couple of things and doesn't add a lot of information. Say something like "For the "update_msg" entry point above, we need a function called update_msg. So take the code below and copy it into your class, below the call function."

Reply:
Rewritten to be more concise.